chart/flux/templates/deployment.yaml

```diff
@@ -195,7 +195,7 @@ spec:
               name: {{ .Values.env.secretName }}
           {{- end }}
           args:
-          {{- if not .Values.clusterRole.create }}
+          {{- if and (not .Values.clusterRole.create) ( .Values.allowedNamespaces) }}
```

Comment:
Testing this by hand with `helm template`

With a values of:
```
clusterRole:
  create: false
allowedNamespaces: []
```

there is no `--allow-namespaces` string present at all in the template output. It is not in the Flux daemon's deployment `args`, (this is the state that could not be configured before this change is merged.)

With a value of:
```
clusterRole:
  create: false
allowedNamespaces:
  - abc
  - def
```

the `--allow-namespace` option is as follows:

```
- --k8s-allow-namespace=abc,def,default
```

(since the Flux namespace is `default` in this case as I am installing with the helm chart in the default namespace, I believe this is correct)

In the template output by 1.10.2 Flux Helm Chart version, this outcome is impossible. The allow-namespace would still get passed with the Flux install namespace, something like this:

```
- --k8s-allow-namespace=default
```

The allow-namespace directive is still used when the `clusterRole.create` is false, because it is assumed you will list the namespaces that Flux can access. I think this actually suggests there is another patch needed for your use case, where you have already provided a clusterRole and it need not be created for you by Flux's helm chart.

Not to delay the merging of this patch any longer, since it has been blocked from merging for long enough.

Reply:
I think I have that a bit backwards, with `clusterRole.create` set to false, allow-namespace can be used, but with `clusterRole.create` set to true, allow-namespace is never used. This behavior is the same as before, as Flux can apparently assume that when a ClusterRole is created, it will have unfettered access to all namespaces.

I'm not sure this is correct. Mulling it over out loud while I decide whether this can merge as-is or if it needs something additional. We can always patch it again, but I'd rather be sure I'm not breaking anything at all inside of the Helm chart, as I'm sure you'll understand

Reply:
I guess it makes sense. You can create a clusterRole that is allowed access to only certain namespaces, but Flux won't create one. Flux only creates a clusterRole that has unfettered access. So it would be wrong to change that, as it will be a breaking change for people who have that configuration. I'm going to leave it alone.

My judgment is this PR can merge as-is 👍 LGTM